csm/core/agent/api.py

```diff
@@ -249,31 +249,63 @@ def getMatchingEndpoint(endpoint_map, path):
         else:
             Log.debug(f"Feature endpoint is not found for {request.path}")
 
+    @classmethod
+    def _extract_bearer(cls, headers) -> str:
+        """
+        Extract the bearer token from HTTP headers.
+
+        :param headers: HTTP headers.
+        :returns: bearer token.
+        """
+
+        hdr = headers.get(CsmAuth.HDR)
+        if not hdr:
+            raise CsmNotFoundError(f'No {CsmAuth.HDR} header')
+        auth_pair = hdr.split(' ')
+        if len(auth_pair) != 2:
+            raise CsmNotFoundError(f'The header is incorrect. Expected "{CsmAuth.HDR} session_id"')
+        auth_type, session_id = auth_pair
+        if auth_type != CsmAuth.TYPE:
+            raise CsmNotFoundError(f'Invalid auth type {auth_type}')
+        return session_id
+
+    @classmethod
+    async def _validate_bearer(cls, login_service, session_id):
+        """
+        Validate the bearer token.
+
+        Search for the session with ID equal to the token value and return it.
+        Throw a Permission denied exception otherwise.
+        :param login_service: login service.
+        :param session_id: bearer token's value.
+        :returns: session object.
+        """
+
+        try:
+            session = await login_service.auth_session(session_id)
+            Log.info(f'Username: {session.credentials.user_id}')
+        except CsmError as e:
+            raise CsmNotFoundError(e.error())
+        if not session:
+            raise CsmNotFoundError('Invalid auth token')
+        return session
+
     @classmethod
     @web.middleware
     async def session_middleware(cls, request, handler):
```

Comment:
Not part of your changes, but this class is full of methods that could be static.  All dependencies on `cls` here are methods that can be turned static as well.  So this method can become a static one, I believe.

Reply:
I've updated the session_middleware that I touched and the related methods. I'll leave the rest out of this story's scope.